src/http/request.cc

```diff
@@ -98,6 +98,12 @@ void request::write_body(const sstring& content_type, noncopyable_function<futur
     this->body_writer = std::move(body_writer);
 }
 
+void request::write_body(const sstring& content_type, size_t len, noncopyable_function<future<>(output_stream<char>&&)>&& body_writer) {
+    set_content_type(content_type);
+    content_length = len;
+    this->body_writer = std::move(body_writer);
```

Comment:
but it doesn't seem like `write_body` actually writes the body...
it looks like it just sets up the body_writer but it doesn't use it.

Reply:
Well, yes. I's named after `request::write_body()` which behaves the same and has been there for 5+ years already :man_shrugging: